src/Bicep.Core/Syntax/SyntaxHierarchy.cs

```diff
@@ -45,12 +45,19 @@ public ParentTrackingVisitor(Dictionary<SyntaxBase, SyntaxBase?> parentMap)
 
             protected override void VisitInternal(SyntaxBase node)
             {
-                var parent = currentParents.Count <= 0 ? null : currentParents.Peek();
-                parentMap.Add(node, parent);
+                try
+                {
+                    var parent = currentParents.Count <= 0 ? null : currentParents.Peek();
+                    parentMap.Add(node, parent);
 
-                currentParents.Push(node);
-                base.VisitInternal(node);
-                currentParents.Pop();
+                    currentParents.Push(node);
+                    base.VisitInternal(node);
+                    currentParents.Pop();
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e.Message);

```

Comment:
>Console [](start = 20, length = 7)

Why is this needed?

Also, we can't have uncontrolled writes to console. SyntaxHierarchy gets invoked in the language server and we're currently using stdout for LSP communication.

Reply:
Good catch. This was checked in by accident. I added the try catch to debug the dup key exception.

Reply:
Maybe we should look into something like https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Microsoft.CodeAnalysis.BannedApiAnalyzers.md#rs0030-do-not-used-banned-apis - although I don't think there's an equivalent in the default dotnet 5 analyzers

Reply:
Thanks for the link. The configuration is quite simple. I added `BannedSymbols.txt` to banned `Console.Write` and `Console.WriteLine`.